packages/dd-trace/src/appsec/rule_manager.js

```diff
@@ -134,6 +153,11 @@ function updateWafFromRC ({ toUnapply, toApply, toModify }) {
     config.apply_state = newApplyState
     if (newApplyError) config.apply_error = newApplyError
   }
+
+  if (newActions.modified) {
+    blocking.updateBlockingConfiguration(concatArrays(newActions).find(action => action.id === 'block'))
+    appliedActions = newActions
+  }
```

Comment:
you still need to apply the new applyState here since it's not done by the batch thing above.
Also wouldn't updating actions when the waf fails cause synchronisation problems ? waf rules are referencing an action, so if you update the actions but not the waf rules then you might end of with an inconsistent state no ?

Reply:
I don't think so.
actions thing is not directly related with the WAF, the waf result would be the same, block or not to block, but the action to do with the result is different.
We don't need to say anything to the waf, we just need to change the behaviour of the blocking.
block with redirect, with custome message, with different status code etc.

Reply:
>you still need to apply the new applyState here since it's not done by the batch thing above.